src/pages/cli/graphql/custom-business-logic.mdx

```diff
@@ -0,0 +1,544 @@
+
+```text
+GET /posts/POST_ID/comments
+Host: www.example.com
+```
+
+Note that there is no check to ensure that the reference variable (here the post ID) exists. When using this technique, it is recommended to make sure the referenced field is non-null.
+
+### How it works
+Definition of `@http` directive:
+
+```graphql
+directive @http(method: HttpMethod, url: String!, headers: [HttpHeader]) on FIELD_DEFINITION
+enum HttpMethod { PUT POST GET DELETE PATCH }
+input HttpHeader {
+  key: String
+  value: String
+}
+```
+
+The `@http` transformer will create one HTTP datasource for each identified base URL. For example, if multiple HTTP resolvers are created that interact with the "https://www.example.com" endpoint, only a single datasource is created. Each directive generates one resolver. Depending on the definition, the appropriate `body`, `params`, and `query` input types are created. Note that `@http` transformer does not support calling other AWS services where Signature Version 4 signing process is required.
+
+
+## VTL resolver
+
+TODO
+
+## Override Amplify-generated resolvers
+
+Amplify generates [AWS AppSync pipeline resolver](https://docs.aws.amazon.com/appsync/latest/devguide/pipeline-resolvers.html) for your queries and mutations. The resolvers are listed the following API resource's folder `amplify/backend/api/<resource_name>/build/resolvers/`. 
+
+![Resolver folders structure in Amplify project folder](/images/cli/resolver_folder_structure.png)
+
+To override an Amplify-generated resolver:
+1. Find the resolver file name you want to override under `build/resolvers`
+2. Place a `.vtl` with the same file name the resource's `resolvers/` (not under `build/`)
+3. Upon the next `amplify api gql-compile` or `amplify push` the Amplify-generated resolver file will be replaced with your overwritten resolver file
+
+```console
+amplify/backend/api/<resource_name>
+├── build
+│   ├── ...
+│   ├── resolvers
+│   │   ├── ...
+│   │   ├── Query.searchTodos.req.vtl # Find resolver file name
+│   │   └── ...
+|   ...
+├── resolvers
+│   └── Query.searchTodos.req.vtl # Place resolver overrides with the same file name here
+```
+
+The example above shows how the `Query.searchTodos.req.vtl` is overwritten with a custom resolver. 
+
+## Extend Amplify-generated resolvers
+
+Amplify generates [AWS AppSync pipeline resolvers](https://docs.aws.amazon.com/appsync/latest/devguide/pipeline-resolvers.html) for your queries and mutations. You can "slot" in your custom business logic between Amplify-generated resolvers. You can find Amplify-generated resolvers under your API resources' `build/resolvers/` folder. The resolver functions file name determines its placement within the slot sequence.
+
+```console
+File name convention: 
+  [Query|Mutation|Subscription].[field name].[slot name].[slot placement].[req|res].vtl
+Example:
+  Mutation.createTodo.postAuth.1.req.vtl
+```
+
+To extend an Amplify-generated resolver:
```

Comment:
Can we a reference to the resolver programming guide?
https://docs.aws.amazon.com/appsync/latest/devguide/resolver-mapping-template-reference-programming-guide.html